src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -116,23 +125,138 @@ protected PSArgument[] CreateVirtualMachineScaleSetCreateOrUpdateParameters()
+
+            // get image
+            var image = Images
+                .Instance
+                .Select(osAndMap =>
+                    new { OsType = osAndMap.Key, Image = osAndMap.Value.GetOrNull(ImageName) })
+                .First(osAndImage => osAndImage.Image != null);
+
+            var isWindows = image.OsType == "Windows";
+            BackendPort = BackendPort
+                ?? (image.OsType == "Windows" ? new[] { 3389, 5985 } : new[] { 22 });
+            
+            var resourceGroup = ResourceGroupStrategy.CreateResourceGroupConfig(ResourceGroupName);
+            
+            var publicIpAddress = resourceGroup.CreatePublicIPAddressConfig(
+                name: PublicIpAddressName,
+                domainNameLabel: DomainNameLabel,
+                allocationMethod: AllocationMethod);
+            
+            var virtualNetwork = resourceGroup.CreateVirtualNetworkConfig(
+                name: VirtualNetworkName, addressPrefix: VnetAddressPrefix);
+
+            var subnet = virtualNetwork.CreateSubnet(SubnetName, SubnetAddressPrefix);
+
+            var loadBalancer = resourceGroup.CreateLoadBalancerConfig(
+                name: LoadBalancerName,
+                froontendPoolName: FrontendPoolName,
+                backendPoolName: BackendPoolName,
+                zones: Zone,
+                publicIPAddress: publicIpAddress,
+                subnet: subnet);
+
+            var frontendIpConfiguration = loadBalancer.CreateFrontendIPConfiguration(
+                name: FrontendPoolName,
+                zones: Zone,
+                publicIPAddress: publicIpAddress,
+                subnet: subnet);
+
+            var backendAddressPool = loadBalancer.CreateBackendAddressPool(
+                name: BackendPoolName);
+
+            var virtualMachineScaleSet = resourceGroup.CreateVirtualMachineScaleSetConfig(
+                name: VMScaleSetName,
+                subnet: subnet,
+                frontendIpConfigurations: new[] { frontendIpConfiguration },
+                backendAdressPool: backendAddressPool,
+                isWindows: isWindows,
+                adminUsername: Credential.UserName,
+                adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
+                image: image.Image,
+                vmSize: VmSize,
+                instanceCount: InstanceCount,
+                upgradeMode: (MyInvocation.BoundParameters.ContainsKey("UpgradePolicyMode") == true ) ? UpgradePolicyMode : (UpgradeMode?) null);
+
+            var client = new Client(DefaultProfile.DefaultContext);
+
+            var current = virtualMachineScaleSet
+                .GetStateAsync(client, new CancellationToken())
+                .GetAwaiter()
+                .GetResult();
+
+            if (Location == null)
             {
-                if (ShouldProcess(this.VMScaleSetName, VerbsCommon.New))
+                Location = current.GetLocation(virtualMachineScaleSet);
```

Comment:
How about adding ```?? "eastus" ``` to this assignment and removing the if staement below?

Reply:
this is in parity with New-AzureRmVM and will be restructured later when we will implement default value per environment.